catalog/core/catalog-core-downloadaction/src/main/java/org/codice/ddf/catalog/download/action/ResourceDownloadActionEndpoint.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.download.action;
+
+import java.lang.management.ManagementFactory;
+
+import javax.management.JMX;
+import javax.management.MBeanException;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.codice.ddf.catalog.resource.download.DownloadToCacheException;
+import org.codice.ddf.catalog.resource.download.ResourceDownloadMBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * REST endpoint class used to manage metacard product downloads.
+ */
+@Path("/")
+public class ResourceDownloadActionEndpoint {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceDownloadActionEndpoint.class);
+    
+    public static final String CONTEXT_PATH = "/internal/catalog/download/cache";
+
+    public static final String SOURCE_PARAM = "source";
+
+    public static final String METACARD_PARAM = "metacard";
+    
+    private final ResourceDownloadMBean resourceDownloadMBean;
+    
+    public ResourceDownloadActionEndpoint() {
+        this.resourceDownloadMBean = createResourceDownloadMBeanProxy();
+    }
+    
+    /**
+     * Starts an asynchronous download of a specific metacard product to the product cache.
+     *
+     * @param sourceId   ID of the federated source where the product should be downloaded from
+     * @param metacardId ID of the metacard that contains the product to download
+     * @return response object containing successful start of download message
+     * @throws DownloadToCacheOnlyException
+     */
+    @GET
+    public Response downloadToCache(@FormParam(SOURCE_PARAM) String sourceId,
```

Comment:
Since this endpoint is related to the download manager and shouldn't really care about the cache, do you think we should use `copyToLocalSite` (or something without `cache` in it) instead? Comment probably applies to some of the other classes and methods that don't care about the cache.